pkg/restore/restic_restore_action.go

```diff
@@ -166,6 +166,10 @@ func getImage(log logrus.FieldLogger, config *corev1.ConfigMap) string {
 		// tagged image name
 		log.Debugf("Plugin config contains image name with tag")
 		return image
+	case len(parts) == 3:
```

Comment:
This is making me think that we may need to switch to using 1+ regexes here for proper matching. A couple of examples that may not properly be matched here:

- `my-custom-registry.io:an-invalid-port-because-its-not-numeric/velero:latest` (shouldn't be valid because the "port" is not numeric, but will be matched by the third case)
- `my-custom-registry.io:8888/velero` (should be a valid tagless image name, i.e. matched by the first case, but will be matched by the second case, not the first)

Reply:
Agree with you. Not really sure how to use regexp in GO properly, though )

Reply:
OK, here's what I'm thinking we should do:

What we're really trying to do with this piece of code is detect if the user has specified an image without a tag (e.g. `steveheptio/velero`), and if so, add a version-specific tag to it, so it becomes e.g. `steveheptio/velero:v1.1.0`.

Instead of splitting `image` on colons (`:`), we could split on front slashes (`/`). If the *last* part (i.e. `parts[len(parts)-1]`) does not contain a colon, it's a tag-less image name, so we want to add the tag by calling/returning `initContainerImage(image)`. Otherwise, we can just use `image` as-is.

If we approach it this way, it should avoid the issue you're running into entirely.

Reply:
i worked it out with '/', but now I get this errror after starting container - 
`An error occurred: some backup storage locations are invalid: error getting backup store for location "default": unable to locate ObjectStore plugin named velero.io/aws`
Where should I look to fix it?

Reply:
It sounds like you've merged/rebased with our latest code, where the cloud-provider plugins are no longer in-tree. The quick solution is to run `velero plugin add velero/velero-plugin-for-aws:v1.0.0` to add the AWS plugin to your deployment.